adapters/repos/db/lsmkv/bucket_recover_from_wal.go

```diff
@@ -47,71 +49,62 @@ func (b *Bucket) recoverFromCommitLogs(ctx context.Context) error {
 			continue
 		}
 
-		if filepath.Join(b.dir, fileInfo.Name()) == b.active.path+".wal" {
-			// this is the new one which was just created
-			continue
-		}
-
 		walFileNames = append(walFileNames, fileInfo.Name())
 	}
 
-	if len(walFileNames) == 0 {
-		// nothing to recover from
-		return nil
-	}
-
 	// recover from each log
 	for _, fname := range walFileNames {
+		path := filepath.Join(b.dir, strings.TrimSuffix(fname, ".wal"))
+
+		cl, err := newCommitLogger(path)
+		if err != nil {
+			return errors.Wrap(err, "init commit logger")
+		}
+		defer cl.close()
+
+		cl.pause()
+		defer cl.unpause()
+
+		mt, err := newMemtable(path, b.strategy, b.secondaryIndices, cl, b.metrics)
```

Comment:
So, If I understand, you are creating a Memtable per WAL and creating a segment per WAL, white the previous code was recovering all WALs into the current active Memtable and flushing it after into the same segment.

Reply:
in practical terms it may be equivalent as only one wal file may be present...